SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp

```diff
@@ -0,0 +1,598 @@
+				const bool bEnabled = false;
+				ProcessCustomActorProperties(TestActor_CustomDisabled, bWrite, bEnabled);
+			}
+		}
+
+		if (AssertTrue(TestActor_AutonomousOnly->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_AutonomousOnly - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bAutonomousExpected = true;
+			const bool bSimulatedExpected = true;
+			ProcessAutonomousOnlyActorProperties(bWrite, bAutonomousExpected, bSimulatedExpected);
+		}
+
+		if (AssertTrue(TestActor_PhysicsEnabled->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_PhysicsEnabled - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bPhysicsEnabled = true;
+			const bool bPhysicsExpected = true;
+			ProcessPhysicsActorProperties(TestActor_PhysicsEnabled, bWrite, bPhysicsEnabled, bPhysicsExpected);
+		}
+
+		if (AssertTrue(TestActor_PhysicsDisabled->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_PhysicsDisabled - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bPhysicsEnabled = false;
+			const bool bPhysicsExpected = true; // Gets replicated through simulated status
+			ProcessPhysicsActorProperties(TestActor_PhysicsDisabled, bWrite, bPhysicsEnabled, bPhysicsExpected);
+		}
+
+		FinishStep();
+	});
+
+	FSpatialFunctionalTestStepDefinition CheckOnOwningClientStepDefinition(/*bIsNativeDefinition*/ true);
+	CheckOnOwningClientStepDefinition.StepName = TEXT("ASpatialTestReplicationConditions Validate Test Actors On Owning Client");
+	CheckOnOwningClientStepDefinition.TimeLimit = 5.0f;
+	CheckOnOwningClientStepDefinition.NativeIsReadyEvent.BindLambda([this]() -> bool {
+		return ActorsReady();
+	});
+	CheckOnOwningClientStepDefinition.NativeStartEvent.BindLambda([this]() {
+		AssertFalse(TestActor_Common->HasAuthority(), TEXT("This client shouldn't have authority"));
+		AssertFalse(TestActor_CustomEnabled->HasAuthority(), TEXT("This client shouldn't have authority"));
+		AssertFalse(TestActor_CustomDisabled->HasAuthority(), TEXT("This client shouldn't have authority"));
+		AssertFalse(TestActor_AutonomousOnly->HasAuthority(), TEXT("This client shouldn't have authority"));
+		AssertFalse(TestActor_PhysicsEnabled->HasAuthority(), TEXT("This client shouldn't have authority"));
+		AssertFalse(TestActor_PhysicsDisabled->HasAuthority(), TEXT("This client shouldn't have authority"));
+
+		AssertTrue(TestActor_Common->IsOwnedBy(GetLocalFlowController()->GetOwner()), TEXT("This client should own the actor"));
+		AssertTrue(TestActor_AutonomousOnly->IsOwnedBy(GetLocalFlowController()->GetOwner()), TEXT("This client should own the actor"));
+		AssertTrue(TestActor_PhysicsEnabled->IsOwnedBy(GetLocalFlowController()->GetOwner()), TEXT("This client should own the actor"));
+		AssertTrue(TestActor_PhysicsDisabled->IsOwnedBy(GetLocalFlowController()->GetOwner()), TEXT("This client should own the actor"));
```

Comment:
Only 4?

Reply:
Only 4 of the actors need the be possessed for the tests. I don't think it would hurt to possess all though, so can make that change if you prefer.